### DISCUSSION TITLE
How to set rtp port range?

### DISCUSSION THREAD

**Discussion by mattiaspihl** [sipsorcery-org/sipsorcery#439] [Q&A] [answered]
Hi,

How do I set what port range should be used for rtp channel?

Best regards,
/Mattias

**Comment by sipsorcery** [comment, marked as answer]
There's currently no option to set the port range used by `RTPSession`. You can pass a single port to the `RTPSession` constructor if you have an exact port you want to use.

The reason the port range was removed was to switch to using each Operating Systems' Ephemeral port range. This removed the need for some tricky logic for things like:
 - Tracking which ports had been used (or recently used) and wrapping around,
 - Allocating ports in a pseudo random fashion to avoid a malicious party being able to easily find in use ports,
 - Bring in line with how other applications, such as WebRTC browsers, allocate port.

**Reply by mattiaspihl** [reply]
Ok good to know.
Thanks for the quick answer and for an awesome library!
